src/deepMap.ts

```diff
@@ -0,0 +1,88 @@
+export class DeepMapEntry<T> {
+  private root: Map<any, any>
+  private closest: Map<any, any>
+  private closestIdx: number = 0
+  isDisposed = false
+
+  constructor(private base: Map<any, any>, private args: any[]) {  
+    let current: undefined | Map<any, any> = this.closest = this.root = base
+    let i = 0
+    for(; i < this.args.length -1; i++) {
+      current = current!.get(args[i])
+      if (!current)
```

Comment:
what do you think about reverting the condition? 
IMHO `if true else false` will be much easier to the next dev to touch this file

Reply:
Done